src/v/redpanda/admin_server.cc

```diff
@@ -1045,19 +1045,51 @@ void config_multi_property_validation(
     if (updated_config.cloud_storage_enabled()) {
         // The properties that cloud_storage::configuration requires
         // to be set if cloud storage is enabled.
-        std::vector<std::reference_wrapper<
-          const config::property<std::optional<ss::sstring>>>>
-          properties{
-            std::ref(updated_config.cloud_storage_secret_key),
-            std::ref(updated_config.cloud_storage_access_key),
-            std::ref(updated_config.cloud_storage_region),
-            std::ref(updated_config.cloud_storage_bucket)};
+        using config_properties_seq = std::vector<std::reference_wrapper<
+          const config::property<std::optional<ss::sstring>>>>;
+
+        config_properties_seq properties{};
+
+        if (
+          updated_config.cloud_storage_credentials_source
+          == model::cloud_credentials_source::config_file) {
+            properties = {
+              std::ref(updated_config.cloud_storage_region),
+              std::ref(updated_config.cloud_storage_bucket),
+              std::ref(updated_config.cloud_storage_access_key),
+              std::ref(updated_config.cloud_storage_secret_key),
+            };
+        } else {
+            properties = {
+              std::ref(updated_config.cloud_storage_region),
+              std::ref(updated_config.cloud_storage_bucket),
+            };
+        }
+
         for (auto& p : properties) {
             if (p() == std::nullopt) {
                 errors[ss::sstring(p.get().name())]
                   = "Must be set when cloud storage enabled";
             }
         }
+
+        if (
+          updated_config.cloud_storage_credentials_source
+          != model::cloud_credentials_source::config_file) {
+            // Credentials are set to be read from cloud API, reject setting
```

Comment:
In principle I like this kind of strictness, but it might cause an issue if @nicolaferraro has already updated the operator to set sts + some placeholder keys, as the updated redpanda will reject the config that used to work.

Can either remove the check or sync with Nicola to make sure it won't cause issues.

Reply:
I think I will remove this check for now, I have created #6865 for a stricter check throughout the system, will add this check there after discussing with @nicolaferraro

Reply:
I have removed the check for now and adjusted the ducktape tests.